src/lib/miradorViewer.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import deepmerge from 'deepmerge';
+
+/** Default Mirador instantiation */
+export default function ({
```

Comment:
Did `MiradorViewer` need to be refactored to accomplish this? Can you talk a little bit about that need if so?

Reply:
* The processing of plugins works different from the current implementation. a) The `plugins` parameter is an array of objects, b) plugins will be stored within the application. Because this is a little simpler in some regards, the `processPlugins` function becomes unnecessary.

* The `MiradorViewer` pretends to be a class, but it doesn't return a instance, but rather a plain object. So, it's actually a function and the refactoring reflects that.

* For the sake of testing I decoupled the default function from it's dependencies to other mirador modules. The init function now has the task to clue things together. I think that's the very notion of an init function.

Reply:
I'm trying to understand the necessity of this refactor to accomplish the plugin goals. I recognize that:

> The MiradorViewer pretends to be a class, but it doesn't return a instance, but rather a plain object. So, it's actually a function and the refactoring reflects that.

Yet does it this refactor need to happen extensively to accomplish your plugin goals?

I'm trying to understand the smallest possible changes needed here to facilitate our plugin system. So that later in the future we can see the gradual change.

> The plugins parameter is an array of objects

Could `MiradorViewer` just take that argument also?

Reply:
Yes, it's possible with less changes. Will change it arccodingly and treat the refactoring in a separate step.